src/ipyradiant/basic_tools/object_literal_collapsing.py

```diff
@@ -0,0 +1,184 @@
+        and a button that automatically moves certain predicates to the 'selected things' side via
+        a set of queries and set analysis.
+
+        This applciation inherits from the ipywidgets.VBox class as well.
+
+        params:
+        :graph: an rdflif.graph.Graph object, representing the graph to be queried on
+        :namespaces: a python dict object that represents the list of namespaces one wishes to use
+        with the current graph.
+        """
+        if graph is not None:
+            kwargs["graph"] = graph
+        if "layout" not in kwargs:
+            kwargs["layout"] = {
+                "min_height": "400px",
+                "flex": "1",
+            }
+        super().__init__(*args, **kwargs)
+
+        for key, value in self.namespaces.items():
+            self.graph.namespace_manager.bind(key, value)
+
+        self.lit_button.on_click(self.populate_predicates)
+        self.children = [self.multiselect, self.lit_button]
+
+    @T.default("graph")
+    def make_default_graph(self):
+        return Graph()
+
+    @T.default("lit_button")
+    def make_default_button(self):
+
+        layout = W.Layout(width="300px", height="40px")  # set width and height
+
+        button = W.Button(
+            description="Add predicates where object is a literal",
+            disabled=False,
+            layout=layout,
+        )
+
+        return button
+
+    @T.observe("graph")
+    def _update_multiselect(self, change):
+        self.multiselect = self.make_default_multiselect()
+
+    @T.default("multiselect")
+    def make_default_multiselect(self):
+        self.all_preds = list(set(list(self.graph.predicates())))
+        self.all_predicates = []
+        for pred in self.all_preds:
+            self.all_predicates.append(
+                CustomURIRef(uri=pred, namespaces=self.graph.namespace_manager)
+            )
+        return MultiPanelSelect(
+            data=self.all_predicates,
+            left_panel_text="Available Predicates",
+            right_panel_text="Predicates to Collapse",
+        )
+
+    def populate_predicates(self, b):
+        """
+        This is a class method that is activated when the 'add predicate where object is a literal' button
+        is pressed. It runs two queries - one to find all predicates where object is not a literal and one to find
+        all predicates where the object is a literal, and then takes the set difference between the two.
+
+        This difference is then converted into what should go to each side of the multiselect, and subsequently
+        passed to the CustomURIRef class in order to get a 'pretty' representation of the URIRef.
+        """
+        # query for all predicates where object is not literal
+        predicates_to_uris = """
+            SELECT DISTINCT  ?p
+            WHERE {
+                    ?s ?p ?o .
+                FILTER (!isLiteral(?o))
```

Comment:
all commands (incl. `FILTER`) should be indented to the same level as the query. No need to update this right now, but just a message for the future.